pkg/windows/windows.go

```diff
@@ -91,6 +95,10 @@ var (
 		kubeProxyServiceName,
 		hybridOverlayServiceName,
 		kubeletServiceName}
+	// OptionalServices is a list of optional Windows services installed by WMCO on some cluster configurations
+	OptionalServices = []string{
```

Comment:
As there is only one OptionalService, it might make more sense to not add this slice, and just export azureCloudNodeManagerServiceName by capitalizing it.

Since it needs to be checked in the same way the RequiredServices is checked here, only in the case of Azure, not having it in a slice would make the logic easier: https://github.com/openshift/windows-machine-config-operator/blob/69a4fee1046d880f55d58b907958d8f75b17643a/test/e2e/validation_test.go#L417-L422

For that test case you can do something like: 
```
if tc.CloudProvider.GetType() == config.AzurePlatform {
   t.Run(svcName, func(t *testing.T) {
					require.Contains(t, svcs, windows.AzureCloudNodeManagerServiceName, "service not found")
					assert.Equal(t, "Running", AzureCloudNodeManagerServiceName)
}
```

Reply:
There is no config access there, so we can't define the platform type the way you mentioned. We also need to check if CCM is enabled, because otherwise we don't need the cloud node manager service.
Do you want me to implement these checks anyway?

Reply:
You can see an example of it done here: https://github.com/openshift/windows-machine-config-operator/blob/7b42724cec23a6c40adff1e6e744276d53a6d4be/test/e2e/create_test.go#L119

you're right, hecking if CCM is enabled would be necessary.

its also important the negative case, where we are ensuring that the service is not running either when platform != azure || CCM is not enabled. We don't want a situation where we are mistakenly running the service on a non-azure platform, where it may be silently running in the background.